test/core.controller.tests.js

```diff
@@ -879,4 +879,59 @@ describe('Chart.Controller', function() {
 			expect(chart.tooltip._options).toEqual(jasmine.objectContaining(newTooltipConfig));
 		});
 	});
+
+	describe('event handling', function() {
```

Comment:
I'm not sure what should be moved exactly, I guess everything that touch the DOM and canvas element.